evidence/verify.go

```diff
@@ -167,50 +166,40 @@ func VerifyLightClientAttack(
 func VerifyDuplicateVote(e *types.DuplicateVoteEvidence, chainID string, valSet *types.ValidatorSet) error {
 	_, val := valSet.GetByAddress(e.VoteA.ValidatorAddress)
 	if val == nil {
-		return fmt.Errorf("address %X was not a validator at height %d", e.VoteA.ValidatorAddress, e.Height())
+		return ErrAddressNotValidatorAtHeight{Address: e.VoteA.ValidatorAddress, Height: e.Height()}
 	}
 	pubKey := val.PubKey
 
 	// H/R/S must be the same
 	if e.VoteA.Height != e.VoteB.Height ||
 		e.VoteA.Round != e.VoteB.Round ||
 		e.VoteA.Type != e.VoteB.Type {
-		return fmt.Errorf("h/r/s does not match: %d/%d/%v vs %d/%d/%v",
-			e.VoteA.Height, e.VoteA.Round, e.VoteA.Type,
-			e.VoteB.Height, e.VoteB.Round, e.VoteB.Type)
+		return ErrDuplicateEvidenceHRTMismatch{*e.VoteA, *e.VoteB}
```

Comment:
Minor: any reason why you prefer to pass the `e.VoteA`/`e.VoteB` by value? (you're copying the struct)

Reply:
It's probably best given that the usage of the error could take place in another goroutine. Technically the original error was copying the `Height`, `Round` and `Type` values.

While `e.VoteA` and `e.VoteB` shouldn't change, there's no guarantee of that unfortunately.